acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/AllowlistWithDnsPersistorAcceptanceTest.java

```diff
@@ -67,22 +71,48 @@ public void setUp() throws Exception {
     cluster.start(this.node);
   }
 
-  @Ignore("test is failing in CI")
   @Test
-  public void manipulatedNodesAllowlistWithHostnameShouldWorkWhenDnsEnabled() {
+  public void addingEnodeWithIp_andThenAddingSameEnodeWithHostname_shouldThrow() {
 
-    node.verify(perm.addNodesToAllowlist(ENODE_ONE, ENODE_TWO));
+    node.verify(perm.addNodesToAllowlist(ENODE_LOCALHOST_IP));
     node.verify(
         perm.expectPermissioningAllowlistFileKeyValue(
-            ALLOWLIST_TYPE.NODES, tempFile, ENODE_ONE, ENODE_TWO));
+            ALLOWLIST_TYPE.NODES, tempFile, ENODE_LOCALHOST_DNS));
+
+    // expect an exception whe adding using hostname, since this node is already added with IP
+    final Condition condition = perm.addNodesToAllowlist(ENODE_LOCALHOST_DNS);
+    assertThatThrownBy(() -> node.verify(condition)).isInstanceOf(RuntimeException.class);
+  }
 
-    node.verify(perm.removeNodesFromAllowlist(ENODE_ONE));
+  @Test
+  public void addingEnodeWithHostNameShouldWorkWhenDnsEnabled() {
+
+    node.verify(perm.addNodesToAllowlist(ENODE_LOCALHOST_DNS));
+
+    // This should just work since there is no IP address to resolve to a host name.
+    // With DNS enabled, the ENODE with the DNS hostname in it should remain as is.
     node.verify(
-        perm.expectPermissioningAllowlistFileKeyValue(ALLOWLIST_TYPE.NODES, tempFile, ENODE_TWO));
+        perm.expectPermissioningAllowlistFileKeyValue(
+            ALLOWLIST_TYPE.NODES, tempFile, ENODE_LOCALHOST_DNS));
+  }
+
+  @Test
+  public void manipulatedNodesAllowlistWithHostnameShouldWorkWhenDnsEnabled() {
+
+    node.verify(perm.addNodesToAllowlist(ENODE_LOCALHOST_DNS, ENODE_TWO_IP));
+    // use DNS config to resolve the Enode with IP. It either resolves to a hostname or remain as is
+    final EnodeURL enodeURL0 =
+        EnodeURLImpl.fromString(
+            ENODE_TWO_IP,
+            ImmutableEnodeDnsConfiguration.builder().dnsEnabled(true).updateEnabled(true).build());
```

Comment:
is this for branch coverage of resolving an enode IP address with name based resolver?

Reply:
yep I added some debug logs to the ticket above. 
it was being resolved to 
`enode://5f8a80d14311c39f35f516fa66...92a0@ip-192-168-0-10.ec2.internal:1234`